client/api/python/heketi/heketi.py

```diff
@@ -44,13 +52,156 @@ def _set_token_in_header(self, headers, method, uri):
         token = jwt.encode(claims, self.key, algorithm='HS256')
         headers['Authorization'] = 'bearer ' + token
 
+        return headers
+
 
     def hello(self):
         method = 'GET'
         uri = '/hello'
 
         headers={}
-        self._set_token_in_header(headers, method, uri)
+        self._set_token_in_header(method, uri)
         r = requests.get(self.host + uri, headers=headers)
         return r.status_code == requests.codes.ok
 
+
+    def _make_request(self, method, uri, data={}):
+        self.method = method
+        self.uri = uri
+        self.data = data
+
+        headers = self._set_token_in_header(self.method, self.uri)
+
+        ''' Ref: http://docs.python-requests.org/en/master/_modules/requests/api/#request '''
+        r = requests.request(self.method,
+                             self.host + self.uri,
+                             headers=headers,
+                             data=json.dumps(data))
+
+        r.raise_for_status()
+
+
+        if r.status_code == requests.codes.accepted and \
+           r.headers['location']:
+            return self._get_queued_response(r.headers['location'])
+
+        elif r.status_code == requests.codes.SEE_OTHER and \
+             r.headers['location']:
+            return self._get_queued_response(r.headers['location'])
+
+        elif r.status_code == requests.codes.created or \
+             r.status_code == requests.codes.ok:
+            if self.method == 'DELETE':
+                return True
+            else:
+                return r.json()
+
+        elif r.status_code == requests.codes.NO_CONTENT:
+            return True
+
+
+
+    def _get_queued_response(self, queue_uri):
```

Comment:
This function returns multiple formats.  Sometimes a request object, sometimes a boolean.  Functions should be setup to return a single expected type.